src/clj/game/cards/ice.clj

```diff
@@ -1456,6 +1456,33 @@
                    :prompt "Choose an ICE to install from HQ"
                    :effect (req (corp-install state side target (zone->name (first (:server run))) {:no-install-cost true}))}]}
 
+   "Formicary"
+   {:req (req (and (:run @state)
+                   (zero? (:position run))
+                   (not (contains? run :corp-phase-43))
+                   (not (contains? run :successful))))
+    :effect (effect (resolve-ability
```

Comment:
Let's move the `:optional` into the root ability of the card instead of the resolve-ability.